src/plist.c

```diff
@@ -728,6 +753,10 @@ Obj TypePlistHom (
         type = CALL_2ARGS( TYPE_LIST_HOM,
             family, INTOBJ_INT(tnum-T_PLIST_HOM+1) );
         ASS_LIST( types, tnum-T_PLIST_HOM+1, type );
+#ifdef HPCGAP
+        // read back element before returning it, in case another thread raced us
+	type = ELM0_LIST( types, tnum-T_PLIST_HOM+1 );
```

Comment:
the indention is fine if you set the tab width to 8. But yeah, this file is horrible in that it wildly mixes tabs and spaces. But if I fix that, the real changes get drowned in tons of tab->space changes. So I'd prefer to do it in a later PR.